### ISSUE TITLE
Add custom icons from within the app 

### ISSUE THREAD

**Issue by Panja0** [raivo-otp/ios-application#67]
It would be great if we can search through DuckDuckGo (or any other search engine) for icons.
I'm not designer savvy to create my own SVG's so just searching for them would be great.

**Comment by mcx808** [comment]
I would also like to see this. Some service providers don't have SVGs available (and some services are internal), but I just want to add in a png so I can differentiate it at a glance.

**Comment by tijme** [comment]
Agreed. This might be added in a future release. However, it can take a while to implement. Especially because I can't host all these icons myself. I need to find a way to upload these to the synchronisation provider, as you would want them synced across all your devices.

To be continued :)

**Comment by PathfinderKP** [comment]
> Agreed. This might be added in a future release. However, it can take a while to implement. Especially because I can't host all these icons myself. I need to find a way to upload these to the synchronisation provider, as you would want them synced across all your devices.
> 
> To be continued :)

What if for now the synced data would include the pictures (what you upload to iCloud)?
Also, another solution (or even method) could be to use links. For me one of the missing icons is for the Bethesda Launcher and if I could add a link (that the app would use) then this link for the Twitter profile could work:
https://pbs.twimg.com/profile_images/1410702902218330120/_cnCYSdG_400x400.png
This way the hosting could be solved as well because the app would sync or even download the pic using the link.

**Comment by tijme** [comment]
Good ideas @PathfinderKP.

Including the picture in the iCloud sync can be done. But I have to look at the limits that can be used for free, as I don't want to pay. Using an external URI would be an idea too. However, I'm not too big of a fan of loading external third-party URI's into a one-time password manager.

**Comment by mcx808** [comment]
> I'm not too big of a fan of loading external third-party URI's into a one-time password manager.

Agreed. Opens up an unnecessary attack surface to something that prioritises security.

I wouldn't even mind the uploaded pngs being local-only if it's easier. It was an annoyance using authy that images had to be set on every device, but at least then it can be done.

**Comment by PathfinderKP** [comment]
PNGs being locally stored would be fine by me as well (maybe even the best and easiest solution as well).
Unless people have to set up Raivo regularly and/or they have a lot of tokens I don't think this feature would be bad without syncing. They set it up once and they might not even have to do it again for months or years.

**Comment by Panja0** [comment]
Storing locally is (of course) fine with me as well!
I don't need to set Raivo multiple times a week. ;-)

**Comment by technicalvampire** [comment]
Local storage and iCloud storage both seem like great options to provide! I understand it can cost money to host for users and that's not always a feasible option.
I moved from Android to iOS and miss Aegis more than any other app. One of my favorite features was the community icon support.
https://aegis-icons.github.io/
https://github.com/aegis-icons/aegis-icons
Beautiful, monochrome circular icons and they are very responsive to icon requests and also take community contributions. Being able to use this set with any TOTP app has been wonderful.
I personally have all of the icons I use hosted in my iCloud account so they are accessible anywhere. So if Raivo could just link to these files somehow, that would be awesome.

**Comment by tijme** [comment]
@technicalvampire agreed. By the way, you know Raivo also has community icon support right? (https://github.com/raivo-otp/issuer-icons/#readme)

**Comment by jkernsjr** [comment]
@tijme I see how this is a similar request to the one I opened and why you closed mine. I also understand trying to come up with a solution that does not cost anything. Question: Are you not able to pull the FavIcon from the URL of the site or something similar, when adding in a new site to Raivo? How do apps like "OTP Auth" do it and remain free?

Thanks for the nice program!

**Comment by technicalvampire** [comment]
@jkernsjr If I recall, OTP Auth unlocks custom icons if you pay a one-time fee of approx. $4 USD. Probably hosted via AWS, but that's just a blind guess.
